dev_scripts/qa.py

```diff
@@ -802,6 +813,31 @@ def _consume_stdin(self):
         while msvcrt.kbhit():
             msvcrt.getch()
 
+    @QABase.task(
+        f"Install the latest version of Python {PYTHON_VERSION_STR}", ref=REF_BUILD
+    )
+    def install_python(self):
+        cur_version = list(sys.version_info[:3])
+
+        logger.info("Getting latest Python release")
+        with urllib.request.urlopen(EOL_PYTHON_URL) as f:
```

Comment:
Maybe factorize this as a separate function? I think it will help readability.

Reply:
Good call, I think I've simplified this in b63c1f4f5e1eaf23baebeec7243b977dd9dc1667.